blockprod/src/detail/mod.rs

```diff
@@ -66,7 +69,141 @@ pub struct JobKey {
 }
 
 pub struct JobHandle {
-    cancel_signal: oneshot::Sender<()>,
+    cancel_sender: UnboundedSender<()>,
+}
+
+pub struct JobManager {
+    chainstate_handle: ChainstateHandle,
+    new_job_sender: UnboundedSender<(Id<GenBlock>, UnboundedSender<()>)>,
+    stop_job_sender: UnboundedSender<(Option<JobKey>, oneshot::Sender<bool>)>,
+    all_jobs: BTreeMap<JobKey, JobHandle>,
+}
```

Comment:
When you're done with your impl, please move this to `job.rs`

Reply:
Yep cool, once it was all going, moving it was the last step.